### PR TITLE
ciacco.add mailquotalimit.list

### PR DESCRIPTION
**Pull request by stephdl**
We can display a list of user sorted by mail quota limit 

NethServer/dev#6622

### CODE REVIEW

**Comment by stephdl** [comment]
@gsanchietti @edospadoni @andre8244  I have no right to set you as reviewers

File: beatrice/public/i18n/en/mailquotalimit-list.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"title":"Mailbox quota limit by user"}
```
Comment (andre8244) [inline beatrice/public/i18n/en/mailquotalimit-list.json:1]: Is the italian file missing in `beatrice/public/i18n/it/mailquotalimit-list.json `? Italian title can be "Quota mailbox utilizzata per utente"

**Review by andre8244 — CHANGES_REQUESTED** [review]

File: beatrice/public/i18n/en/mailquotalimit-list.json (hunk 1 above)
Comment (gsanchietti) [reply beatrice/public/i18n/en/mailquotalimit-list.json:1]: Correct, @stephdl could you please add also such file?

File: ciacco/miners/mailquotalimit-list [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/perl
+
+#
+# Copyright (C) 2021 Nethesis S.r.l.
```
Comment (gsanchietti) [inline ciacco/miners/mailquotalimit-list]:
```suggestion
# Copyright (C) 2022 Nethesis S.r.l.
```

File: ciacco/miners/mailquotalimit-list [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+my $quota_status = $cdb->get_prop('dovecot', 'QuotaStatus') || 'disabled';
+if ($quota_status eq 'enabled' && -f '/usr/libexec/nethserver/mail-quota') {
+    my $quota = decode_json(`/usr/libexec/nethserver/mail-quota`);
+    foreach (keys %$quota) {
+        my $size = int($quota->{$_}{'size'} || 0);
+        my $max = int($quota->{$_}{'max'} || 0);
+        next if ($size == 0);
+        next if ($max == 0);
+        my $quotaLimit = int(($size/$max)*100);
+        push(@data, {name => $_, count => $quotaLimit});
```
Comment (gsanchietti) [inline ciacco/miners/mailquotalimit-list]: The script already return a `perc` field, we can use it.

Output example:
```
  "giacomo@nethesis.it": {
    "perc": 9,
    "msg": 114453,
    "max": 20971520,
    "size": 2062054
  },

```

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: ciacco/miners/mailquotalimit-list (hunk 2 above)
Comment (stephdl) [reply ciacco/miners/mailquotalimit-list]: I still need a bit of time to understand the year change, sorry

File: ciacco/miners/mailquotalimit-list (hunk 3 above)
Comment (stephdl) [reply ciacco/miners/mailquotalimit-list]: good, done

File: beatrice/public/i18n/en/mailquotalimit-list.json (hunk 1 above)
Comment (stephdl) [reply beatrice/public/i18n/en/mailquotalimit-list.json:1]: yes, done

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**